File: cli/po-cli/src/services/financerequestservice.ts
import { FinanceRequestService as BaseFinanceRequestService, HttpService, Party } from 'wt-common-cli';

export class FinanceRequestService extends BaseFinanceRequestService {

    constructor(httpService: HttpService) {
        super(httpService);
    }

    async createFinanceRequest(requester: Party, financierIds: string[], purchaseOrderId: string, amount: number, interest: number, monthLength: number) {
        const financeRequestBody = {
            requester,
            financierIds,
            purchaseOrderId,
            amount,
            interest,
            monthLength
        };

        const partialFrg = await this.httpService.post('financerequests', financeRequestBody);

        return this.getFinanceRequestsByGroupHash(partialFrg.hash);
    }

    async withdrawFinanceRequest(reqId: string) {
        return this.httpService.put(`financerequests/${reqId}/withdraw`, {});
    }

    async acceptFinanceRequest(reqId: string) {
        return this.httpService.put(`financerequests/${reqId}/accept`, {});
    }
}

File: po-rest/src/main/java/org/tradefinance/main/services/FinanceRequestService.java
package org.tradefinance.porest.services;

import java.util.Collection;

import org.tradefinance.assets.FinanceRequest;
import org.tradefinance.assets.FinanceRequestGroup;
import org.tradefinance.assets.defs.Party;

public interface FinanceRequestService extends org.tradefinance.common.services.FinanceRequestService {

    public Collection<FinanceRequest> getFinanceRequests(String behalfOfId) throws Exception;

    public FinanceRequestGroup createFinanceRequest(Party requester, String[] financierIds, String purchaseOrderId, double amount, double interest, int monthLength) throws Exception;

    public void acceptFinanceRequest(String id) throws Exception;

    public void withdrawFinanceRequest(String id) throws Exception;
}
